### PR TITLE
fix(navbar): revamp style and texts

### PR DESCRIPTION
**Pull request by FredericEspiau**
## Revamp Navbar
MBE-188

Related to:
- ovh-ux/ovh-manager-web#872
- ovh-ux/ovh-manager-cloud#1237
- ovh-ux/ovh-manager-telecom#1214

### CODE REVIEW

**Comment by jleveugle** [comment]
Hello @FredericEspiau,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

File: client/app/components/navbar/navbar.service.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  class Navbar {
+    static buildMenuHeader(text) {
+      return `
```
Comment (Jisay) [inline client/app/components/navbar/navbar.service.js]: Why do you build html into js service?

File: client/app/components/navbar/navbar-notification.service.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class NavbarNotificationService {
   constructor(
     $interval, $q, $translate,
-    Alerter, atInternet, constants, OvhApiNotificationAapi,
+    Alerter, atInternet, constants, navbar, OvhApiNotificationAapi,
```
Comment (frenautvh) [inline client/app/components/navbar/navbar-notification.service.js]: is navbar is a class it should be uppercased

File: client/app/components/navbar/navbar.service.js (hunk 1 above)
Comment (frenautvh) [reply client/app/components/navbar/navbar.service.js]: i don't understand either, if you really want to do that, using $compile et importing the template would looks better

**Comment by ovh-cds** [comment]
CDS Report build-manager#666.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✔ 
* Stage 2
  * Build manager CA ✔ 
  * Build manager EU ✘  
  * Build manager US ✔ 
  * Tests ✔

File: client/app/components/user/session/user-session.service.js [hunk 3]
```diff
@@ -406,141 +414,148 @@ class SessionService {
...
-    return {
-      name: 'user',
-      title: currentUser.firstName,
-      iconClass: 'icon-user',
-      nichandle: currentUser.nichandle,
-      fullName: `${currentUser.firstName} ${currentUser.lastName}`,
-      subLinks: [
+    const useNewText = ['FR'].includes(currentUser.ovhSubsidiary);
+
+    const textTitle = useNewText
```
Comment (FredericEspiau) [inline client/app/components/user/session/user-session.service.js]: Bit weird to have the condition done twice, you could:

```suggestion
    const p = useNewText ? this.Navbar.buildMenuHeader(`${this.$translate.instant('otrs_menu_userAccount_1', { username: currentUser.firstName })}<br>${this.$translate.instant('otrs_menu_userAccount_2')}`) : this.$q.when(currentUser.firstName)
```

File: client/app/components/navbar/navbar.less [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  align-self: center;
+
+  .medium({
+    font-size: rem-calc(10);
+  });
+}
+
+.transition() {
+  display: inline-block;
+  transition: all 0.2s ease-in;
```
Comment (antleblanc) [inline client/app/components/navbar/navbar.less]: Using `all` has a cost regarding the animation frame budget.

The only property you have to care about in your case is `transform`.

File: client/app/components/navbar/navbar.service.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+{
+  class Navbar {
+    constructor($compile, $rootScope, $timeout) {
+      this.$compile = $compile;
+      this.$rootScope = $rootScope;
+      this.$timeout = $timeout;
+    }
+
+    buildMenuHeader(content) {
+      const compiledTemplate = this.$compile(template)(angular.merge(
```
Comment (FredericEspiau) [inline client/app/components/navbar/navbar.service.js]:
```suggestion
      const compiledTemplate = this.$compile(template)({
      ...this.$rootScope.$new(true),
      { $ctrl: { content } }
      })
```
?

**Comment by marie-j** [comment]
@antleblanc I think @FredericEspiau can answer better than me but the arrows were required but there are a too specific need to figure in `oui-navbar`

File: client/app/components/navbar/navbar.service.js (hunk 5 above)
Comment (marie-j) [reply client/app/components/navbar/navbar.service.js]: This won't work as it seems `prototype` is not copied

File: client/app/components/navbar/navbar.service.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      this.$timeout = $timeout;
+    }
+
+    buildMenuHeader(content) {
+      const compiledTemplate = this.$compile(template)(_.merge(
+        this.$rootScope.$new(true),
+        { $ctrl: { content } },
+      ));
+      // $timeout is required in order to let angular's scope $digest
+      return this.$timeout(() => compiledTemplate.html());
```
Comment (frenautvh) [inline client/app/components/navbar/navbar.service.js]: not mandatory but a sighly better way : 
$timeout will do a digest for every single scope starting from the rootScope
you only need to digest the scope used for compilation
```
$compile(myTemplate)(myScope);
myScope.$digest();
```

**Comment by jleveugle** [comment]
Hello @FredericEspiau,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request closed without merging**